### PR TITLE
fix(server): add security levels to policies

### LINKED ISSUE #6169: Server Does Not Publish Security Levels
The server does not populate the security level of the endpoint security policies, except to use zero for deprecated policies and one for the rest. This does not allow the client to score the relative merits of the policies. I have coded a fix that adds a `securityLevel` member to the security policy structure, and picks this up when creating endpoints, rather than basing the decision on the policy names. I have also modified `UA_ServerConfig_addAllSecureEndpoints` to similarly rely on the security level when deciding whether a policy is secure.

## Checklist

Please provide the following information:

 - [X] open62541 Version (release number or git tag): v1.3.8-1478-g08d2cbeca
 - [ ] Other OPC UA SDKs used (client or server): 
 - [ ] Operating system:
 - [ ] Logs (with `UA_LOGLEVEL` set as low as necessary) attached
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ ] Critical issue

### PR DESCRIPTION
**Pull request by MikeGranby**
Report security levels to allow clients to select best security policy.

Closes #6169

### CODE REVIEW

**Comment by jpfr** [comment]
Cool! Merged.

**Comment by mcpat** [comment]
Hi @MikeGranby and @jpfr 

I do not want to warm up an already merged PR. Still there are questions about the values for the securityLevel which are now applied:

- [Aes128-Sha256-RsaOaep](https://profiles.opcfoundation.org/profile/2058) is targeted at applications "... with average security needs" (and uses SHA-1) -> gets securityLevel 30 assigned.
- [Basic256Sha256](https://profiles.opcfoundation.org/profile/2059) is targeted at applications "... with high security needs" (and also uses SHA-1) -> gets securityLevel 10 assigned.
- [Aes256-Sha256-RsaPss](https://profiles.opcfoundation.org/profile/2060) is targeted at applications "... with a need for high security" -> gets securityLevel 20 assigned.

To summarise: I am not really convinced about the values being assigned here. Could you please explain the intention?

**Comment by MikeGranby** [comment]
Yeah, I must admit those values were based on my assumptions as to relative levels of security, and based on what you’ve quoted there, are incorrect.

Aes128 should be 10, and then presumably Basic256 with 20 and Aes256 with 30?

Sorry about that!

Mike Granby
Granby Consulting LLC
+1 (717) 309-5635<tel:+1%20(717)%20309-5635> mobile
+1 (717) 287-1958<tel:+1%20(717)%20287-1958> office
www.granbyconsulting.com<http://www.granbyconsulting.com/>

**Comment by mcpat** [comment]
This sounds reasonable. I will create a PR that does this change.
